skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationRunManagerListener.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold
+
+import com.google.container.tools.core.analytics.UsageTrackerProvider
+import com.google.container.tools.skaffold.metrics.SKAFFOLD_DEV_CONFIGURATION_ADDED
+import com.google.container.tools.skaffold.metrics.SKAFFOLD_SINGLE_RUN_CONFIGURATION_ADDED
+import com.google.container.tools.skaffold.run.SkaffoldDevConfiguration
+import com.google.container.tools.skaffold.run.SkaffoldSingleRunConfiguration
+import com.intellij.execution.RunManagerListener
+import com.intellij.execution.RunnerAndConfigurationSettings
+import com.intellij.openapi.components.ProjectComponent
+import com.intellij.openapi.project.Project
+
+/**
+ * Project component that connects itself to project message bus when any project opens and uses
+ * [RunManagerListener] events to catch additions of new Skaffold run configurations to send
+ * client side pings.
+ */
+class SkaffoldConfigurationRunManagerListener(val project: Project) : ProjectComponent {
+    override fun projectOpened() {
+        project.messageBus.connect().subscribe(RunManagerListener.TOPIC,
```

Comment:
I mean to put the above comment here, sorry

Reply:
For project message bus it should be taken care automatically since this is a disposable and once project is closed, all listeners should be gone - at least this is what we see in IDE code itself for project message bus.